sql/srv_session.cc

```diff
@@ -841,6 +846,14 @@ bool Srv_session::open() {
     return true;
   }
 
+#ifdef WITH_WSREP
+  if (thd.wsrep_cs().state() != wsrep::client_state::s_none) {
+    thd.wsrep_cs().cleanup();
+  }
+  thd.wsrep_client_thread = true;
```

Comment:
Shouldn't we lock LOCK_wsrep_thd before changing this? wsrep_sync_wait depends on it

Reply:
Question not answered.